intranet/apps/events/views.py

```diff
@@ -24,29 +24,34 @@ def events_view(request):
 
     Shows a list of events occurring in the next week, month, and
     future.
-
     """
 
     is_events_admin = request.user.has_admin_permission("events")
 
     if request.method == "POST":
         if "approve" in request.POST and is_events_admin:
-            event_id = request.POST.get("approve")
-            event = get_object_or_404(Event, id=event_id)
-            event.rejected = False
-            event.approved = True
-            event.approved_by = request.user
-            event.save()
-            messages.success(request, "Approved event {}".format(event))
+            event_id = get_id(request.POST.get("approve"))
+            if event_id:
```

Comment:
Nitpicking here, but I would do:
```suggestion
            if event_id is not None:
```
Since `get_id()` will always either return an int or `None`. I know that ids should never be 0, but this feels a little better.